geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/io/Sizable.scala

```diff
@@ -0,0 +1,40 @@
+/***********************************************************************
+ * Copyright (c) 2013-2020 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.utils.io
+
+import com.typesafe.scalalogging.LazyLogging
+import org.ehcache.sizeof.SizeOf
+import org.ehcache.sizeof.impl.PassThroughFilter
+
+import scala.util.control.NonFatal
+
+trait Sizable {
+
+  /**
+   * Approximate size in memory, in bytes
+   *
+   * @return
+   */
+  def calculateSizeOf(): Long
+}
+
+object Sizable extends LazyLogging {
+
+  private val sizeOf = try { SizeOf.newInstance() } catch {
+    case NonFatal(e) =>
+      logger.warn("Could not load SizeOf, memory estimates will be off:", e)
+      new SizeOf(new PassThroughFilter(), true, true) {
```

Comment:
Under what conditions would we expect this to happen?!

Reply:
I don't think this would necessarily ever happen... you can see the different approaches attempted by ehcache:sizeof here: https://github.com/ehcache/sizeof#the-different-engines
this was more best practice of not killing the whole system if for some reason it can't load a sizeof implementation.